src/main/java/StringCalculator.java

```diff
@@ -0,0 +1,59 @@
+import java.util.regex.Pattern;
+
+
+/**
+ * 문자열을 숫자로 변환하여 합산하는 계산기임. <br>
+ * 쉼표 또는 콜론을 기본 구분자로 사용함.<br>
+ * //<구분자>\n 해당 위치에 원하는 구분자를 넣으면,<br>
+ * 단일 문자로 커스텀 구분자 지정이 가능함.<br>
+ * //@\n1:2@3,4 같은 형식도 처리 가능<br>
+ * 음수나 숫자가 아닌 값이 포함되면 예외를 발생시킴.
+ */
+public class StringCalculator {
+
+    public static int add(String input) {
+        if (input.isEmpty()) {
+            return 0;
+        }
+
+        String delimiter = "[,|:]"; // 기본 구분자 정의
+
+        if (input.startsWith("//")) { // 커스텀 구분자 처리 (여러 글자 지원)
+            int delimiterEnd = input.indexOf("\n");
+            String customDelimiter = input.substring(2, delimiterEnd);
+            delimiter = "[,|:]|" + Pattern.quote(customDelimiter);
+            input = input.substring(delimiterEnd + 1);
+        }
+
+        String[] numbers = input.split(Pattern.compile(delimiter).pattern());
+        return sum(numbers);
```

Comment:
[디미터의 법칙](https://tecoble.techcourse.co.kr/post/2020-06-02-law-of-demeter/)을 아시나요?

준석은 어느 쪽이 더 이해하기 쉬운가요? 😄
```suggestion
        Pattern pattern = Pattern.compile(delimiter);
        String[] numbers = pattern.split(input);
        return sum(numbers);
```